assets/questions/bomberman.js

```diff
@@ -0,0 +1,147 @@
+                    up[c] += 1
+                else:
+                    memo[r][c] += left + up[c]
+        right = 0
+        down = [0 for x in range(cols)]
+        for r in range(rows - 1, -1, -1):
+            right = 0
+            for c in range(cols - 1, -1, -1):
+                if board[r][c] == "e":
+                    right += 1
+                    down[c] += 1
+                else:
+                    memo[r][c] += right + down[c]
+                    if memo[r][c] > maxEnemy:
+                        maxEnemy = memo[r][c]
+        return maxEnemy
+    @classmethod
+    def bombAllEnemiesOnBoard(cls, board):
+        if (len(board) == 0 or len(board[0]) == 0): return 0
+        count = 0
+        for r in range(len(board)):
+            for c in range(len(board[0])):
+                if board[r][c] == "e":
+                    count += 1
+        return count
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'For this question, you will be given a borad, in which all elements are string. ',
+        'X represent a wall, e represent an enemy, "" represent an empty cell. ',
+        'You can place an bomb in an empty cell, which will kill enemies in vertical and horizontal ',
+        'direction until it met a wall. ',
+        'Please write a function bomb(board), to find the best position to place a bomb in order to',
+        'kill as many enemies as possible, then return the number of enemies killed.'
+      ].join('')
+    ],
+    prerequisiteSkills: ['Arrays', 'Strings', 'String Manipulation'],
+    acquiredSkills: ['String Manipulation'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'bomb',
+    correctnessTests: [
+      {
+        input: [],
+        allowedOutputs: [0]
+      },
+      {
+        input: [[], [], []],
+        allowedOutputs: [0]
+      },   
+      {
+        input: [["e", "", "e", "x", "", "e"]],
+        allowedOutputs: [2]
+      },
+      {
+        input: [["e", "", "e", "", "", "e"]],
+        allowedOutputs: [3]
+      },
+      {
+        input: [["e"], ["x"], [""], ["x"]],
+        allowedOutputs: [0]
+      },
+      {
+        input: [["e"], ["e"], [""], ["x"]],
+        allowedOutputs: [2]
+      },
+      {
+        input: [["", "e", ""], ["e", "", "e"], ["", "e", ""]],
+        allowedOutputs: [4]
+      }
+    ],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.ignoreWalls',
+      messages: [[
+        "You cannot kill an enemy behind a wall x. ",
+        "Stop your fire once it meets a wall."
+      ].join('')
+      ]
+    },
+    {
+      buggyFunctionName: 'AuxiliaryCode.bombAllEnemiesOnBoard',
+      messages: [[
+        "You cannot simply kill all enemies on the board. ",
+        "Your bomb can only kill enemies in the same row and column. ",
+        "Stop your fire once it meets a wall."
+      ].join('')
+      ]
+    },
+    {
+      buggyFunctionName: 'AuxiliaryCode.negative',
```

Comment:
I do not think this is a useful test case -- do you expect that many students will attempt to submit this as a valid solution?